iosxr_iso2vbox.py

```diff
@@ -95,43 +97,53 @@ def main(argv):
     copy_to_artifactory = False
     create_ova = False
     verbose = False
-    master_opts = '==> iosxr_iso2vbox.py [-i <Input ISO>] [-s <Remote Input ISO>] [-o, --ova], [-a, --artifactory] [-v, --verbose] [-h, --help]'
 
-    # Suck in the input ISO and handle errors
-    try:
-        opts, args = getopt.getopt(argv, 'hva:oi:s:', ['help', 'verbose', 'artifactory=', 'ova', 'iso=', 'scp='])
-    except getopt.GetoptError:
-        print(master_opts)
-        sys.exit(2)
-    for opt, arg in opts:
-        if opt in ('-h', '--help'):
-            print('==> A tool to create a VirtualBox box from an ISO')
-            print(master_opts)
-            print('==> E.g. iosxr-xrv64-vbox/iosxr_iso2vbox.py -iso iosxrv-fullk9-x64.iso')
-            print('==> E.g. iosxr-xrv64-vbox/iosxr_iso2vbox.py -scp server:<path>/iosxrv-fullk9-x64.iso')
-            sys.exit()
-        elif opt in ('-i', '--iso'):
-            input_iso = arg
-        elif opt in ('-s', '--scp'):
-            iso_to_scp = arg
-            cmdstring = 'scp %s@%s .' % (getpass.getuser(), iso_to_scp)
-            print('==> Will attempt to scp the remote image to current working dir. You may be required to enter your password.')
-            time.sleep(2)
-            print('==> %s\n' % cmdstring)
-            subprocess.call(cmdstring, shell=True)
-            input_iso = os.path.basename(iso_to_scp)
-        elif opt in ('-a', '--artifactory'):
-            copy_to_artifactory = True
-            reason = arg
-            if not reason:
-                reason = 'No reason for update specified'
-        elif opt in ('-v', '--verbose'):
-            verbose = True
-        elif opt in ('-o', '--ova'):
-            create_ova = True
+    parser = argparse.ArgumentParser(
+        formatter_class=RawTextHelpFormatter,
+        description='A tool to create an IOS XRv Vagrant VirtualBox box from ' +
+        'an IOS XRv ISO.\n' '\nThe ISO will be installed, booted, configured ' +
+        'and unit-tested. \n"vagrant ssh" provides access ' +
+        'to IOS XR Linux global-vrf namespace \nwith internet access.',
+        epilog="E.g.: iosxr-xrv64-vbox/iosxr_iso2vbox.py " +
+        "iosxrv-fullk9-x64.iso -a 'new image' -o -v")
+    parser.add_argument('ISO_FILE',
+                        help='local ISO filename or remote URI ISO filename...')
+    parser.add_argument('-a', '--artifactory', nargs='?',
+                        const='check_string_for_empty',
+                        help='add optional reason for uploading this box')
```

Comment:
`help="Upload box to Artifactory. You can optionally specify a reason for uploading this box."`